extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```
@@ -0,0 +1,326 @@
+          "zookeeper.connect required property"
+      );
+
+      kafkaProperties.setProperty("group.id", factoryId);
+      final String topic = getKafkaTopic();
+      LOG.debug("About to listen to topic [%s] with group.id [%s]", topic, factoryId);
+      final Map<String, String> map = cacheManager.getCacheMap(factoryId);
+      mapRef.set(map);
+      // Enable publish-subscribe
+      kafkaProperties.setProperty("auto.offset.reset", "smallest");
+
+      consumerConnector = buildConnector(kafkaProperties);
+
+      final List<KafkaStream<String, String>> streams = consumerConnector.createMessageStreamsByFilter(
+          new Whitelist(Pattern.quote(topic)), 1, DEFAULT_STRING_DECODER, DEFAULT_STRING_DECODER
+      );
+
+      if (streams == null || streams.isEmpty()) {
+        throw new IAE("Topic [%s] had no streams", topic);
+      }
+      if (streams.size() > 1) {
+        throw new ISE("Topic [%s] has %d streams! expected 1", topic, streams.size());
+      }
+      final KafkaStream<String, String> kafkaStream = streams.get(0);
+      final ConsumerIterator<String, String> it = kafkaStream.iterator();
+
+      final ListenableFuture<?> future = executorService.submit(
+          new Runnable()
+          {
+            @Override
+            public void run()
+            {
+              while (it.hasNext()) {
+                final MessageAndMetadata<String, String> messageAndMetadata = it.next();
+                final String key = messageAndMetadata.key();
+                final String message = messageAndMetadata.message();
+                if (key == null || message == null) {
+                  LOG.error("Bad key/message from topic [%s]: [%s]", topic, messageAndMetadata);
+                  continue;
+                }
+                doubleEventCount.incrementAndGet();
+                map.put(key, message);
+                doubleEventCount.incrementAndGet();
+                LOG.trace("Placed key[%s] val[%s]", key, message);
+              }
```

i agree with @nishantmonu51, you might wanna swallow some of the exceptions if not all.